packages/shared/src/hooks/useReadingRank.ts

```diff
@@ -29,15 +30,27 @@ const defaultRank: MyRankData = {
   reads: 0,
 };
 
+const checkShouldShowRankModal = (rankLastSeen: Date) => {
+  if (!rankLastSeen) {
+    return true;
+  }
+
+  return new Date(rankLastSeen) < new Date();
```

Comment:
This will always be true. We should compare the rank last seen to the last article read time.
Because only when the user reads an article there's progress so as long as the user didn't read anything there's no progress to show. I don't know if we this information on the client but maybe we can return it as part of the user reading rank query.

Reply:
I agree, comparing it to the last read would make more sense and improve the predictability of the scenario. Thanks!